packages/venia-ui/lib/components/ProductOptions/swatch.js

```diff
@@ -43,18 +43,28 @@ const Swatch = props => {
 
     const classes = mergeClasses(defaultClasses, props.classes);
 
-    // TODO: use the colors from graphQL when they become available.
-    //   https://github.com/magento/graphql-ce/issues/196
-    //   https://github.com/magento/pwa-studio/issues/1633
-    const randomColor = memoizedGetRandomColor(value_index);
+    let swatchValue;
+    const isImage = swatch_data && swatch_data.value.charAt(0) !== '#';
+
+    if (isImage) {
+        const imagePath = generateUrlFromContainerWidth(
```

Comment:
So, having provided feedback on image swatch, I'm a bit apprehensive to include it in this scope. If I use one of our 4x5 images this is the result as-is:
<img width="297" alt="Screen Shot 2020-05-05 at 4 26 07 PM" src="https://user-images.githubusercontent.com/462953/81117882-7c2def80-8eed-11ea-8f67-56d8c81ae78c.png">

This is the result if I use the optimized thumbnail from GraphQL:
<img width="291" alt="Screen Shot 2020-05-05 at 4 26 53 PM" src="https://user-images.githubusercontent.com/462953/81117916-88b24800-8eed-11ea-9b0e-c5651c831f23.png">

And finally if I change the implementation to hard-code width/height to the exact dimensions of these tiles (3rem/3rem or 48px/48px):
<img width="295" alt="Screen Shot 2020-05-05 at 4 27 12 PM" src="https://user-images.githubusercontent.com/462953/81118006-b13a4200-8eed-11ea-80ae-f18248de2911.png">

We're making progress on not assuming every images aspect ratio, but a square box now forces us to actually support this natively. Would like the team's thoughts on this; but I'm going to propose we remove support for image swatch until we can support different aspect ratios.

I would be willing to accept this if we put in a comment that says you **must** upload swatch images with a 1x1 aspect ratio, and change implementation to this (only way I could get images to render correctly):
```
const imagePath = generateUrl(swatch_data.thumbnail, 'image-swatch')(48);
```

Reply:
>I'm going to propose we remove support for image swatch until we can support different aspect ratios.

I think that is fair. The initial scope of this ticket was to replace random color with the actual color.

However it does seem like `swatch_data.value` can be an image url. We still need to have a fall-back thing to display.